### DISCUSSION TITLE
Best practice to execute the job now and the next ones with CronTrigger

### DISCUSSION THREAD

**Discussion by dmatton** [quartznet/quartznet#1865] [Q&A]
Hello everyone,

Actually, I have a .NET Worker Service with Quartz.Net, and I configured a CronTigger to execute a job every Monday morning.
If I start the Worker Service now (so Tuesday), I need to wait until next Monday, seems logic. However, on the first start, I would like the Worker Service to execute immediately the Job and the next execution will be on Monday.

What would be the best practice for that kind of execution?

Regards,
David

**Comment by lahma** [comment]
I would probably just call `IScheduler.TriggerJob` on service startup to get immediate fire. Under the hood it create a simple trigger with repeat count of one.
